src/InvoiceManager.php

```diff
@@ -0,0 +1,326 @@
+
+        $invoice = new Invoice();
+        $invoice->user_id = $userId;
+        $invoice->currency_id = $currencyId;
+        $invoice->meta = $meta;
+        if (isset($localizedDetails['title'])) {
+            $invoice->title = $localizedDetails['title'];
+        }
+        $invoice->status = InvoiceStatus::UNPAID->value;
+        $invoice->save();
+        $invoice->products()
+                ->createMany($this->buliderInvoiceProducts($products));
+        $invoice->update([
+                             'amount' => Number::fromInt($invoice->products->sum('total_amount'))
+                                               ->getValue(),
+                         ]);
+
+        return $invoice;
+    }
+
+    public function delete(int $invoiceId): void
+    {
+        // TODO: Implement delete() method.
+
+        $invoice = $this->getInvoiceById($invoiceId);
+        if (InvoiceStatus::PAID == $invoice->status) {
+            throw new InvalidInvoiceStatusException();
+        }
+        if ($invoice->products->isNotEmpty()) {
+            $invoice->products()
+                    ->delete();
+        }
+        if ($invoice->payments->isNotEmpty()) {
+            $invoice->payments()
+                    ->delete();
+        }
+        $invoice->delete();
+    }
+
+    public function update(int $invoiceId, array $changes): IInvoice
+    {
+        // TODO: Implement update() method.
+
+        $invoice = $this->getInvoiceById($invoiceId);
+        if (InvoiceStatus::PAID == $invoice->status) {
+            throw new InvalidInvoiceStatusException();
+        }
+
+        return DB::transaction(function () use ($invoice, $changes) {
+            if (isset($changes['title'])) {
+                $invoice->title = $changes['title'];
+            }
+            if (isset($changes['user_id'])) {
+                $invoice->user_id = $changes['user_id'];
+            }
+            if (isset($changes['meta'])) {
+                $invoice->meta = $changes['meta'];
+            }
+            if (isset($changes['currencyId'])) {
+                $invoice->currency_id = $changes['currencyId'];
+            }
+            if (isset($changes['products'])) {
+                $this->updateInvoiceProduct($changes['products'], $invoice);
+            }
+            $invoice->save();
+            $invoice->update([
+                                 'amount' => Number::fromInt($invoice->products->sum('total_amount'))
+                                                   ->getValue(),
+                             ]);
+
+            return $invoice;
+        });
+    }
+
+    public function addProductToInvoice(int $invoiceId, array $product): IProduct
+    {
+        // TODO: Implement addProductToInvoice() method.
+
+        $invoice = $this->getInvoiceById($invoiceId);
+
+        return DB::transaction(function () use ($invoice, $product) {
+            $record = new Product();
+            if (isset($product['price'])) {
```

Comment:
can we add a product to the invoice without price or title or count or ...?

Reply:
done